src/components/dev-hub/community-champion-form.js

```diff
@@ -98,7 +108,21 @@ const Form = React.memo(({ setSuccess, success }) => {
     const [agreeToEmail, setAgreeToEmail] = useState(false);
     const [canSubmit, setCanSubmit] = useState(true);
     const handleSubmit = async e => {
-        // TODO: implement this
+        e.preventDefault();
+        setCanSubmit(false);
+        const data = {
+            first_name: firstName,
+            last_name: lastName,
+            email,
+            experience,
+            bio,
+            agree_to_email: agreeToEmail,
+        };
+        const callback = hasSuccess => {
+            setCanSubmit(!hasSuccess);
+            setSuccess(hasSuccess);
```

Comment:
am noticing that the order of these two make a difference, calling setSuccess before setCanSubmit results in the following warning: 

![Screen Shot 2021-07-28 at 11 05 47 AM](https://user-images.githubusercontent.com/46455728/127366961-ac89b7cd-6f4f-4e95-80b9-4bfaa181cf90.png)

I believe this is because updating success state causes the modal to re-render and unmounts the form to mount the "Thanks for applying!" success component which results in setCanSubmit being a no-op since form has been unmounted (and canSubmit is one of the form's states). 

I wonder if we're also getting this warning on the project sign up form when we submit since setSuccess is also being called first https://github.com/mongodb/devhub/blob/staging/src/components/dev-hub/project-sign-up-form.js#L57-L58 ...haven't tested it yet since i didn't want to set off the trigger to send an email lol

Reply:
Thanks for calling this out, I think your suggestion removes that possibility since if the component should re-render, only then do we update state. Let's go ahead and add that in! I'll add some more comments to the PR now :)